app/src/androidTestKiwix/java/org/kiwix/kiwixmobile/tests/BasicTest.java

```diff
@@ -53,81 +45,39 @@
   @Rule
   public ActivityTestRule<SplashActivity> mActivityTestRule = new ActivityTestRule<>(
       SplashActivity.class);
+  @Rule
+  public GrantPermissionRule readPermissionRule = GrantPermissionRule.grant(Manifest.permission.READ_EXTERNAL_STORAGE);
+  @Rule
+  public GrantPermissionRule writePermissionRule = GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE);
 
   @Test
   public void basicTest() {
     enterHelp();
 
-    onView(withId(R.id.drawer_layout)).perform(DrawerActions.open());
-
-
-    ViewInteraction textView7 = onView(
-        allOf(withId(R.id.titleText), withText("Help"),
-            childAtPosition(
-                childAtPosition(
-                    withId(R.id.left_drawer_list),
-                    0),
-                0),
-            isDisplayed()));
-    textView7.check(matches(withText("Help")));
-
-    ViewInteraction imageView = onView(
-        allOf(childAtPosition(
-            allOf(withId(R.id.new_tab_button),
-                childAtPosition(
-                    IsInstanceOf.instanceOf(android.widget.LinearLayout.class),
-                    1)),
-            0),
-            isDisplayed()));
-    imageView.check(matches(isDisplayed()));
-
-    ViewInteraction imageView2 = onView(
-        allOf(childAtPosition(
-            allOf(withId(R.id.new_tab_button),
-                childAtPosition(
-                    IsInstanceOf.instanceOf(android.widget.LinearLayout.class),
-                    1)),
-            0),
-            isDisplayed()));
-    imageView2.check(matches(isDisplayed()));
-
-    onView(withId(R.id.drawer_layout)).perform(DrawerActions.close());
+    openDrawer();
+
+    assertDrawerIsOpen();
+
+    assertDisplayed(R.id.titleText);
```

Comment:
This looks like an improvement on the previous code, my one niggle is that it could detect that the text is displayed anywhere on the screen rather than in a particular element (unless it limits the scope of where it searches to that of the open drawer?)
Perhaps we'll be able to find ways to check the content and layout more diligently over time. For now this is a good start.